client/ayon_deadline/plugins/publish/global/collect_jobinfo.py

```diff
@@ -0,0 +1,345 @@
+            deadline_addon.get_machines_by_server_name(deadline_server_name))
+        for machine in machines:
+            cls.machines_enum_values.append(
+                {"value": machine, "label": machine})
+
+    @classmethod
+    def get_attr_defs_for_instance(cls, create_context, instance):
+        """Get list of attr defs that are set in Settings as artist overridable
+
+        Args:
+            create_context (ayon_core.pipeline.create.CreateContext)
+            instance (ayon_core.pipeline.create.CreatedInstance):
+
+        Returns:
+            (list)
+        """
+        if instance.product_type not in cls.families:
+            return []
+
+        host_name = create_context.host_name
+
+        task_name = instance["task"]
+        folder_path = instance["folderPath"]
+        task_entity = create_context.get_task_entity(folder_path, task_name)
+
+        task_name = task_type = None
+        if task_entity:
+            task_name = task_entity["name"]
+            task_type = task_entity["taskType"]
+        profile = filter_profiles(
+            cls.profiles,
+            {
+                "host_names": host_name,
+                "task_types": task_type,
+                "task_names": task_name,
+                # "product_type": product_type
+            }
+        )
+        if not profile:
+            return []
+        overrides = set(profile["overrides"])
+        if not overrides:
+            return []
+
+        defs = [
+            UISeparatorDef("deadline_defs_starts"),
+        ]
+
+        defs.extend(cls._get_artist_overrides(overrides, profile))
+
+        # explicit frames to render - for test renders
+        defs.append(
+            TextDef(
+                "frames",
+                label="Frames",
+                default="",
+                tooltip="Explicit frames to be rendered. (1, 3-4)"
+            )
+        )
+
+        defs = cls._host_specific_attr_defs(create_context, instance, defs)
+
+        defs.append(
+            UISeparatorDef("deadline_defs_end")
+        )
+
+        return defs
+
+    @classmethod
+    def _get_artist_overrides(cls, overrides, profile):
+        """Provide list of all possible Defs that could be filled by artist"""
+        # should be matching to extract_jobinfo_overrides_enum
+        default_values = {}
+        for key in overrides:
+            default_value = profile[key]
+            if key == "machine_limit":
+                filtered = []
+                for value in default_value:
+                    if value in cls.machines_enum_values:
+                        filtered.append(value)
+                default_value = filtered
+            if key == "limit_groups":
+                filtered = []
+                for value in default_value:
+                    if value in cls.limit_group_enum_values:
+                        filtered.append(value)
+                default_value = filtered
+            if isinstance(default_value, list):
```

Comment:
I think that both `machine_limit` and `limit_groups` are lists, and they should stay as list, or?

Reply:
Changed to lists, only serialized to comma separated string